pkg/search/proxy/store/multi_cluster_cache.go

```diff
@@ -98,16 +98,8 @@ func (c *MultiClusterCache) UpdateCache(resourcesByCluster map[string]map[schema
 			newCachedResources[resource] = struct{}{}
 		}
 	}
-	for resource := range c.cachedResources {
-		if _, exist := newCachedResources[resource]; !exist {
-			delete(c.cachedResources, resource)
-		}
-	}
-	for resource := range newCachedResources {
-		if _, exist := c.cachedResources[resource]; !exist {
-			c.cachedResources[resource] = struct{}{}
-		}
-	}
+	c.cachedResources = newCachedResources
```

Comment:
Seems `c.cachedResources` value never be used. Can you replace `map` with `sets.Set` to modify `c.cachedResources` definition?

Reply:
@whitewindmills This part of the code is used.

```
// HasResource return whether resource is cached.
func (c *MultiClusterCache) HasResource(resource schema.GroupVersionResource) bool {
	c.lock.RLock()
	defer c.lock.RUnlock()
	_, ok := c.cachedResources[resource]
	return ok
}
```